homeassistant/components/lastfm/sensor.py

```diff
@@ -52,27 +59,24 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class LastfmSensor(Entity):
     """A class for the Last.fm account."""
 
-    def __init__(self, user, lastfm_api):
+    def __init__(self, hass: HomeAssistantType, user, lastfm_api):
         """Initialize the sensor."""
         self._user = lastfm_api.get_user(user)
         self._name = user
         self._lastfm = lastfm_api
-        self._state = "Not Scrobbling"
+        self._state = STATE_NOT_SCROBBLING
         self._playcount = None
         self._lastplayed = None
         self._topplayed = None
         self._cover = None
 
+        self.entity_id = generate_entity_id(ENTITY_ID_FORMAT, user, hass=hass)
+
     @property
     def name(self):
         """Return the name of the sensor."""
         return self._name
 
-    @property
-    def entity_id(self):
-        """Return the entity ID."""
-        return f"sensor.lastfm_{self._name}"
```

Comment:
Seeing as the username (`self._name`) would be unique, instead of using `generate_entity_id` we could just use `self._name.lower()` in the above line to solve the issue of usernames with capitals, though we would still be assuming that lastfm usernames do not contain any other characters that are not allowed in home assistants entity_ids. That being said `generate_entity_id` is a more complete solution.